core/CMakeLists.txt

```diff
@@ -30,7 +25,14 @@ install(
   DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
 install(
-  DIRECTORY include/utils
+  DIRECTORY include/algorithms
   DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
-add_library(traccc::core ALIAS traccc_core)
\ No newline at end of file
+if (TRACCC_BUILD_cuda)
+  find_library(CUDART_LIBRARY cudart ${CMAKE_CUDA_IMPLICIT_LINK_DIRECTORIES})
+  target_include_directories(traccc_core
+    PUBLIC
+    ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES})
+endif()
```

Comment:
I'm confused about this part. Why do you want to link `libcudart` publicly against `traccc::core`? :confused: At a quick glance I don't see that you would've added any CUDA code to the headers in the core library...

I also don't understand why using `PUBLIC` for the include directory type actually works. :confused: Since `traccc_core` is an `INTERFACE` library, I would've bet money that you would only be allowed to declare include directories and libraries with the `INTERFACE` keyword in `target_include_directories(...)` and `target_link_libraries(...)` for `traccc_core`...

All in all, as long as you just use VecMem types in `traccc_core`, it should be enough to just link it against `vecmem::core`, and not worry about any direct CUDA/HIP/SYCL dependency on this library. (Those explicit dependencies should only be declared for the libraries holding the CUDA/HIP/SYCL specific algorithms.)

Reply:
At least in the PR, we don't need to link libcudart to traccc::core. I just didn't push some codes that are defined with the cuda type qualifiers (`__host__ __device` ), which needs cuda library. For example, in my latest code, the functions `definitions/algebras.hpp` has the cuda type qualifiers to be used inside kernel. It will be useful in case we want to reuse cpu tracking helper function in gpu kernel.

About the INTERFACE keyword, I will make a proper change soon

Reply:
I see. But for just using the `__host__` and `__device__` keywords you should not make the CUDA runtime library's headers visible to the compilation. The CUDA headers have nothing to do with what `__host__` and `__device__` do.

In VecMem we also put these declarations on (template) functions in `vecmem::core`. Without `vecmem::core` including any CUDA headers. That works fine, since when the headers are parsed by `nvcc`, it makes use of these keywords. While the C\+\+ compiler just doesn't see them.

Long story short, you should just use `__device__` and `__host__` without worrying about the build system. Only the code that calls `cuda<Foo>(...)` functions needs to interact with the CUDA header(s).

Reply:
Um. seems you are right after trying to compile without linking libraries. Gonna fix it